nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutFile.java

```diff
@@ -81,6 +86,24 @@
 
     public static final String FILE_MODIFY_DATE_ATTRIBUTE = "file.lastModifiedTime";
     public static final String FILE_MODIFY_DATE_ATTR_FORMAT = "yyyy-MM-dd'T'HH:mm:ssZ";
+    
+    public static final Pattern RWX_PATTERN = Pattern.compile("^([r-][w-])([x-])([r-][w-])([x-])([r-][w-])([x-])$");
+    public static final Pattern NUM_PATTERN = Pattern.compile("^[0-7]{3}$");
+    
+    private static final Validator PERMISSIONS_VALIDATOR = new Validator() {
+        @Override
+        public ValidationResult validate(String subject, String input, ValidationContext context) {
```

Comment:
Since the permission property supports EL, if EL is used, we need to determine the configuration is valid and skip further validation. I'll add that when I merge this.